### PR TITLE
[SPARK-37589][SQL][TESTS] Improve the performance of `PlanParserSuite`

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
https://github.com/apache/spark/pull/34543 provides the parse method `parseQuery`. So we can reduce the time spent on executing `PlanParserSuite` by replace `parsePlan`.

### Why are the changes needed?
Improve the performance of `PlanParserSuite`.

### Does this PR introduce _any_ user-facing change?
'No'.
Just update tests.

### How was this patch tested?
Jenkins test.

### CODE REVIEW

**Comment by beliefer** [comment]
ping @cloud-fan

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Could you give us some improvement numbers in the PR description, @beliefer ?
> So we can reduce the time spent on executing PlanParserSuite by replace parsePlan.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/PlanParserSuite.scala [hunk 1]
```diff
@@ -39,9 +39,16 @@ class PlanParserSuite extends AnalysisTest {
     comparePlans(parsePlan(sqlCommand), plan, checkAnalysis = false)
   }
 
+  private def assertQueryEqual(query: String, plan: LogicalPlan): Unit = {
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/PlanParserSuite.scala:42]: This makes it hard to write tests as people need to pick between `assertEqual` and `assertQueryEqual`.

How much we can speed up  this test suite by doing so?

Comment (beliefer) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/PlanParserSuite.scala:42]: About tens of milliseconds

**Comment by beliefer** [comment]
Maybe it not worth.

**Comment by dongjoon-hyun** [comment]
Thank you for closing, @beliefer .

**Pull request closed without merging**